app/src/org/commcare/provider/IdentityCalloutHandler.java

```diff
@@ -0,0 +1,174 @@
+package org.commcare.provider;
+
+import android.content.Intent;
+
+import org.commcare.android.javarosa.IntentCallout;
+import org.commcare.commcaresupportlibrary.identity.IdentityResponseBuilder;
+import org.commcare.commcaresupportlibrary.identity.model.IdentificationMatch;
+import org.commcare.commcaresupportlibrary.identity.model.MatchResult;
+import org.commcare.commcaresupportlibrary.identity.model.MatchStrength;
+import org.commcare.commcaresupportlibrary.identity.model.RegistrationResult;
+import org.commcare.commcaresupportlibrary.identity.model.VerificationMatch;
+import org.commcare.util.LogTypes;
+import org.javarosa.core.model.FormDef;
+import org.javarosa.core.model.condition.EvaluationContext;
+import org.javarosa.core.model.instance.AbstractTreeElement;
+import org.javarosa.core.model.instance.TreeReference;
+import org.javarosa.core.services.Logger;
+import org.javarosa.core.services.locale.Localization;
+import org.javarosa.core.util.OrderedHashtable;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.util.Collections;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.Vector;
+
+import androidx.annotation.StringDef;
+
+public class IdentityCalloutHandler {
```

Comment:
Is there something unique to how this app works? The pattern of having the class itself consist ~entirely of static members makes it seem like it's doing something specific that I don't quite understand, but I could also see this just being a set of helper methods.

Reply:
I was trying to mimic the `SimprintsCalloutProcessing` here. Both of these can potentially implement an Identity Provider interface though we won't be able to define all these methods as `static` in that case. Would that be ok ? Not sure if `static` methods are of a particular advantage here.